gtsfm/utils/transforms.py

```diff
@@ -0,0 +1,48 @@
+"""Utility functions for geometric transformations.
+
+Authors: Akshay Krishnan
+"""
+from typing import List
+
+import numpy as np
+from gtsam import Unit3
+
+
+def euclidean_to_spherical_directions(directions: List[Unit3]) -> np.ndarray:
+    """Converts a list of Unit3 directions to spherical coordinates (azimuth, zenith).
+    Angles are given in a compass frame:
+    zenith=0 is along +y, azimuth=0 is along -z, and azimuth=pi/2 is along +x.
+    This follows 1DSfM's convention
+    https://github.com/wilsonkl/SfM_Init/blob/f801a4ace3b34f990cbda3c57b96387ce19c90c1/sfminit/onedsfm.py#L132
+
+    Args:
+        directions: List of Unit3 directions.
+
+    Returns:
+        Nx2 numpy array where N is the length of the list and columns are (aximuth, zenith).
+    """
+    directions_array = np.array([d.point3() for d in directions])
+    azimuth = np.arctan2(directions_array[:, 0], -directions_array[:, 2])
+    zenith = np.arccos(directions_array[:, 1])
+    return np.column_stack((azimuth, zenith))
+
+
+def spherical_to_euclidean_directions(spherical_coords: np.ndarray) -> List[Unit3]:
+    """Converts an array of spherical coordinates to a list of Unit3 directions.
+
+    Args:
+        directions: Nx2 array where the first column are [azimuth, zenith].
```

Comment:
@akshay-krishnan by "zenith" do you mean "elevation" or "inclination"?

The zenith is an imaginary point directly "above" a particular location,
https://en.wikipedia.org/wiki/Zenith

I assume you're talking about (theta, phi)?
https://en.wikipedia.org/wiki/Spherical_coordinate_system

Reply:
Yeah, let me update that. Borrowed terminology and direction convention from 1dsfm.

Reply:
updated